File: frontend/src/microservices/operations/airline/objects/ManagerFlightOrderListResponse.ts
// 鏈枃浠跺畾涔?operations 妯″潡鐨?`ManagerFlightOrderListResponse`锛屼綔涓哄垪琛ㄥ搷搴旀暟鎹苟鎻愪緵 JSON 缂栬В鐮併€?

import type { ManagerFlightOrderResponse } from './ManagerFlightOrderResponse'

export type ManagerFlightOrderListResponse = {
  orders: ManagerFlightOrderResponse[]
}

export const managerFlightOrderListResponseFromJson = (json: string): ManagerFlightOrderListResponse =>
  JSON.parse(json) as ManagerFlightOrderListResponse

export const managerFlightOrderListResponseToJson = (value: ManagerFlightOrderListResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/airline/objects/ManagerFlightOrderListResponse.scala
// 这个文件定义 airline 订单列表响应。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerFlightOrderListResponse(orders: List[ManagerFlightOrderResponse])
object ManagerFlightOrderListResponse:
  given sourceEncoder: Encoder[ManagerFlightOrderListResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerFlightOrderListResponse] = deriveDecoder
